nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/test/java/org/apache/nifi/web/search/attributematchers/SearchableMatcherTest.java

```diff
@@ -25,14 +25,17 @@
 import org.apache.nifi.search.SearchContext;
 import org.apache.nifi.search.SearchResult;
 import org.apache.nifi.search.Searchable;
-import org.junit.Before;
-import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
 import org.mockito.Mock;
 import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
 
 import java.util.Collection;
 import java.util.HashSet;
 
+@MockitoSettings(strictness = Strictness.LENIENT)
```

Comment:
There's a lot of `lenient` settings being used with Mockito, is this necessary for all the classes changed in this MR (whether as part of the `@MacktioSettings` annotation on a class or a `@Mock` annotation on a single bean)?

The Mockito default is `strict` rather than `lenient` to try and encourage (what they believe to be) better testing practices, e.g. don't define unnecessary stubs

Reply:
I wouldn't consider it a blocker for a merge. AFAIK it doesn't actually create real problems aside from runtime bloat which isn't a critical issue for us. It's something that we could put in the backlog as a TODO for the whole code base as a cleanup activity.

Reply:
I agree it's not a blocker. If the lenient setting can simply be removed now then let's do that (has it been applied by default during some automated upgrade routine?)

If it'll be more rework of the tests then fine to raise a tidy-up ticket on the backlog

I only raised it because it looks like a change in mock behaviors during the junit 5 upgrade - Mockito has had strict on by default since version 2 and that shouldn't have changed between junit 4 and 5 runners (that I'm aware of, but I could well have missed such a thing)

Reply:
@ChrisSamo632  I am not aware of what you are referring to in reference to Mockito functionality. But with the upgrade to JUnit5 in my cases the unit tests were failing with an exception like the following where it actually suggests using the lenient option which I did to get these unit cases working. 

```
org.mockito.exceptions.misusing.UnnecessaryStubbingException: 
Unnecessary stubbings detected.
Clean & maintainable test code requires zero unnecessary code.
...
...
Please remove unnecessary stubbings or use 'lenient' strictness. More info: javadoc for UnnecessaryStubbingException class.
```

Reply:
@dan-s1 The ideal solution is to remove the unnecessary stubbing instead of using the lenient mocking mode. If it ends up being too involved, it could be revisited as others noted.

Reply:
@exceptionfactory I ended up using `@MockitoSettings(strictness = Strictness.LENIENT)` where there were multiple mocks and it was hard determining which one was causing the exception so I used this more as a "hammer" solution. The ones where I used on  `@Mock(lenient = true)` is where one or two mocks were involved so I can look at those again to see if I can remove the unnecessary stubbing. Does the sound reasonable?

Reply:
Thanks, that sounds good.